addons/website_sale/static/tests/tours/website_sale_shop_editor_tour.js

```diff
@@ -18,3 +18,26 @@ wTourUtils.registerWebsitePreviewTour("shop_editor", {
     extra_trigger: "iframe div.o_pricelist_dropdown a[data-bs-toggle=dropdown][aria-expanded=false]",
     content: "Click on the pricelist again.",
 }]);
+
+wTourUtils.registerWebsitePreviewTour("shop_editor_set_product_ribbon", {
+    test: true,
+    url: "/shop",
+    edition: true,
+}, [{
+    content: "Click on first product",
+    trigger: "iframe .oe_product:first",
+}, {
+    content: "Open the ribbon selector",
+    trigger: ".o_wsale_ribbon_select we-toggler",
+    extra_trigger: ".o_wsale_ribbon_select",
+}, {
+    content: "Select a ribbon",
+    trigger: 'we-button:contains("Sale")',
+}, {
+    content: "Save",
+    trigger: 'button[data-action="save"]',
```

Comment:
Use our `clickOnSave` tour utils to do this, it will be more robust for future changes of this tour